File: day1/p1/solution.rb
def captcha(nums)
  nums.each_with_index.reduce(0) do |sum, (num, i)|
    next_num = i == nums.length - 1 ? nums[0] : nums[i+1]
    sum + (num == next_num ? next_num : 0)
  end
end

nums = File.read('input.txt').split('').map(&:to_i)
p captcha(nums)

File: day1/p1/solution.exs
# This is my first Elixir program, no judging

{:ok, input} = File.read("input.txt")

nums = input
|> String.split("", trim: true)
|> Enum.map(fn(num) -> String.to_integer(num) end)

{result, _} = nums
|> Enum.reduce({0, 0}, fn(num, {agg, i}) ->
  next_num = if (i == length(nums) - 1), do: Enum.at(nums, 0), else: Enum.at(nums, i+1)
  to_sum = if (num == next_num), do: next_num, else: 0
  {agg+to_sum, i+1}
end)

IO.inspect(result)
